libsql-sqlite3/src/wal.h

```diff
@@ -222,10 +221,13 @@ struct libsql_wal {
   unsigned char lockError;            /* True if a locking error has occurred */
   WalIndexHdr *pSnapshot;             /* Start transaction here if not NULL */
   sqlite3 *db;
-  libsql_wal_methods *pMethods;       /* Virtual methods for interacting with WAL */
-  void *pMethodsData;                 /* Optional context for private use of libsql_wal_methods */
+} sqlite3_wal;
+
+struct libsql_wal {
+    libsql_wal_methods methods; /* virtual wal methods */
+    void* self; /* methods receiver */
```

Comment:
right, so it's a pointer-to-implementation. In that case let's name it.

Reply:
I need a translation here 😄